pkg/test/suite/testcase_subscriber.go

```diff
@@ -38,173 +47,144 @@ func (s subscriberTestCase) GetInput() interface{} {
-	for _, sub := range testCases {
-		s.publish(sub)
+func isTestCaseSubscriber(method reflect.Method) bool {
+	if method.Func.Type().NumIn() != 1 {
+		return false
 	}
 
-	app.Stop()
-	app.WaitDone()
-
-	config := s.Env().Config()
-	logger := s.Env().Logger()
-
-	for _, sub := range testCases {
-		configKey := mdlsub.GetSubscriberOutputConfigKey(sub.GetName())
-		outputType := config.GetString(configKey)
-
-		switch outputType {
-		case mdlsub.OutputTypeDb:
-			dbSub, ok := sub.(dbSubscriberTestCase)
-
-			if !ok {
-				s.FailNow("invalid subscription test case", "the test case for the subscription of %s has to be of the db type", sub.GetName())
-				return
-			}
-
-			orm, err := db_repo.NewOrm(config, logger)
-			if err != nil {
-				s.FailNow("can't initialize orm", "the test case for the subscription of %s can't be initialized", sub.GetName())
-			}
-
-			fetcher := &DbSubscriberFetcher{
-				t:    s.T(),
-				orm:  orm,
-				name: sub.GetName(),
-			}
-
-			dbSub.Assert(s.T(), fetcher)
-
-		case mdlsub.OutputTypeDdb:
-			ddbSub, ok := sub.(ddbSubscriberTestCase)
-
-			if !ok {
-				s.FailNow("invalid subscription test case", "the test case for the subscription of %s has to be of the ddb type", sub.GetName())
-				return
-			}
-
-			fetcher := &DdbSubscriberFetcher{
-				t: s.T(),
-				repo: func(model interface{}) (ddb.Repository, error) {
-					return ddb.NewRepository(config, logger, &ddb.Settings{
-						ModelId: ddbSub.ModelIdTarget,
-						Main: ddb.MainSettings{
-							Model:              model,
-							ReadCapacityUnits:  5,
-							WriteCapacityUnits: 5,
-						},
-					})
-				},
-				name: sub.GetName(),
-			}
-
-			ddbSub.Assert(s.T(), fetcher)
-
-		case mdlsub.OutputTypeKvstore:
-			dbSub, ok := sub.(kvstoreSubscriberTestCase)
-
-			if !ok {
-				s.FailNow("invalid subscription test case", "the test case for the subscription of %s has to be of the kvstore type", sub.GetName())
-				return
-			}
-
-			store, err := kvstore.NewConfigurableKvStore(config, logger, sub.GetName())
-			if err != nil {
-				s.FailNow("can't initialize kvStore", "the test case for the subscription of %s can't be initialized", sub.GetName())
-			}
-
-			fetcher := &KvstoreSubscriberFetcher{
-				t:     s.T(),
-				store: store,
-				name:  sub.GetName(),
-			}
-
-			dbSub.Assert(s.T(), fetcher)
-		}
+	if method.Func.Type().NumOut() != 2 {
+		return false
 	}
-}
-
-func (s SubscriberTestSuite) publish(sub SubscriberTestCase) {
-	attrs := mdlsub.CreateMessageAttributes(sub.GetModelId(), "create", 0)
-
-	streamName := fmt.Sprintf("subscriber-%s", sub.GetName())
-	s.Env().StreamInput(streamName).Publish(sub.GetInput(), attrs)
-}
 
-func (s SubscriberTestSuite) DbTestCase(testCase DbSubscriberTestCase) SubscriberTestCase {
-	modelId, err := mdl.ModelIdFromString(testCase.ModelId)
+	var v SubscriberTestCase
+	var subscriberTestCaseType = reflect.ValueOf(&v).Elem().Type()
```

Comment:
What is the difference to `reflect.ValueOf(v).Type()`?

Reply:
that one results in:
```
panic: reflect: call of reflect.Value.Type on zero Value [recovered]
	panic: reflect: call of reflect.Value.Type on zero Value
```

Reply:
Ahh, that is an interface, yeah, makes sense then